### DISCUSSION TITLE
Is there more info about kernel errors?

### DISCUSSION THREAD

**Discussion by tve** [sgenoud/replicad#90] [Q&A]
I frequently get kernel errors. 90% of the time it's because I'm passing the wrong type of object to some function, e.g., `lineTo(x, y)` vs. `lineTo([x, y])` type of stuff. In those cases, I'd love to get some feedback on which call returns the kernel error. In 90% of cases it's a true kernel error and I wonder whether there is any more info. I haven't dug in to see whether the error numbers mean anything and whether I can look them up somewhere... Also, I wonder what the best way would be to get kernel bugs addressed... I've had to stop using fillets 'cause they mostly just produce kernel errors.

**Comment by sgenoud** [comment]
I know that there is a way to [compile the kernel](https://ocjs.org/docs/getting-started/file-size#what-if-thats-still-too-much) such that the errors it returns are more explicit. But it makes the whole bundle bigger and longer to load.

I should have different builds of the kernel for the renders in share mode and in edit mode - but I have never separated them.

> I haven't dug in to see whether the error numbers mean anything and whether I can look them up somewhere..

The numbers linked with the errors do not mean anything as far as I know.

> Also, I wonder what the best way would be to get kernel bugs addressed... I've had to stop using fillets 'cause they mostly just produce kernel errors.

With working a bit with this kernel, I have realised that fillet tends to work better when multiple edges are selected and filleted at the same time [like here](https://replicad.xyz/docs/tutorial-overview/modifications#multiple-radii).

Another tactic is to be careful not having fillets that would take a whole face (making it a tenth of the millimeter smaller for instance).

I have never done it myself, but, as FreeCAD uses the same kernel, they might have some "best practices" to work with fillets that are compatible with the kernel. (a quick search points to [questions going in that direction](https://www.reddit.com/r/FreeCAD/comments/12ikhwh/what_is_the_safest_way_to_add_fillets/)

That said, potentially paying the company that work on [open cascade](https://www.opencascade.com/) itself might be a way to fix them in a better way.

**Reply by sgenoud** [reply]
> wrong type of object to some function, e.g., lineTo(x, y) vs. lineTo([x, y]) 

That said, I could have more helpful errors in replicad for this particular error

**Reply by tve** [reply]
Having different builds would definitely be nice 'cause the "kernel error" messages are one of the more frustrating aspects of the current system.

**Comment by raydeleu** [comment]
Would it be an option to show some kind of console where the progress of the build is shown? At least that makes it possible to detect the location of the statement in the code where the kernel error was triggered. As it is now, finding the error means only changing a few things or even a single thing at a time before trying the code.

**Reply by sgenoud** [reply]
you can look into the standard web console - your `console.log` statements will be shown there!
